service/src/main/java/bio/terra/workspace/service/workspace/flight/cloud/aws/SetWorkspaceApplicationEgressIngressStep.java

```diff
@@ -0,0 +1,118 @@
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ec2.model.AuthorizeSecurityGroupEgressRequest;
+import software.amazon.awssdk.services.ec2.model.AuthorizeSecurityGroupIngressRequest;
+import software.amazon.awssdk.services.ec2.model.IpPermission;
+import software.amazon.awssdk.services.ec2.model.IpRange;
+import software.amazon.awssdk.services.ec2.model.UserIdGroupPair;
+
+public class SetWorkspaceApplicationEgressIngressStep implements Step {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(SetWorkspaceApplicationEgressIngressStep.class);
+  private final CrlService crlService;
+  private final SamService samService;
+  private final AwsCloudContextService awsCloudContextService;
+
+  public SetWorkspaceApplicationEgressIngressStep(
+      CrlService crlService, AwsCloudContextService awsCloudContextService, SamService samService) {
+    this.crlService = crlService;
+    this.samService = samService;
+    this.awsCloudContextService = awsCloudContextService;
+  }
+
+  private void authorizeEgress(EC2SecurityGroupCow regionCow, String securityGroupId) {
+    try {
+      regionCow.authorizeEgress(
+          AuthorizeSecurityGroupEgressRequest.builder()
+              .groupId(securityGroupId)
+              .ipPermissions(
+                  IpPermission.builder()
+                      .fromPort(0)
+                      .toPort(0)
+                      .ipProtocol("-1")
+                      .ipRanges(IpRange.builder().cidrIp("0.0.0.0/0").build())
+                      .build())
+              .build());
+
+      logger.info("Allowed egress to all destinations for Security Group {}", securityGroupId);
+    } catch (CrlEC2DuplicateSecurityRuleException e) {
+      // This is expected, as we may race with creation of a default egress rule, which sets the
+      // same IP permissions.  It may also happen when re-playing the step on failure; this is also
+      // OK as the end result is the same.
+      logger.info("Ignoring duplicate egress rule creation error");
+    }
+  }
+
+  private void authorizeIngress(EC2SecurityGroupCow regionCow, String securityGroupId) {
+    try {
+      regionCow.authorizeIngress(
+          AuthorizeSecurityGroupIngressRequest.builder()
+              .groupId(securityGroupId)
+              .ipPermissions(
+                  IpPermission.builder()
+                      .fromPort(0)
+                      .toPort(0)
```

Comment:
Does this allow all ports, or none? I don't see anything about using `0` in the AWS documentation

Reply:
Yeah, the docs are super sketchy about this, this is the best ref I could find: https://repost.aws/questions/QUVWll2XoIRB6J5JqZipIwZQ/what-is-mean-fromport-is-0-and-toport-is-0-in-security-groups-ippermission-ippermissionegress#ANlQylxlBvSaqrIip2SAFajQ

I'll add some verbiage in comments for egress and ingress being explicit about what these rules accomplish.